ilc/client/TransactionManager.js

```diff
@@ -106,9 +108,15 @@ export class TransactionManager {
         observer.observe(targetNode, { childList: true, subtree: true, attributeFilter: ['style'] });
     };
 
-    #renderFakeSlot = slotName => {
+    #renderFakeSlot = (slotName, appName) => {
         const targetNode = getSlotElement(slotName);
         const clonedNode = targetNode.cloneNode(true);
+
+        const cssBundleLink = document.querySelector(`link[data-fragment-id="${appName}"]`);
```

Comment:
I'm afraid that in case you have same app rendered twice on the page (ex: header/footer) - you would eventually remove links that you shouldn't be touching.

Reply:
Also I don't like that now `handlePageTransaction()` knows about appName. Looks like abstraction leakage a bit